### PR TITLE
Support for specifying project by name, and todo IDs by unique prefix

### PR DESCRIPTION
**Pull request by mthowe**
Just started using todoist CLI today and these are 2 minor things that were causing me a bit of pain so I went ahead and added them.

Big Disclaimer: I am not a golang programmer, so some or all of this may not be the idiomatic Go way of doing things. It might also break some encapsulation or otherwise that you were hoping to maintain. But I've tested both changes and they work for me.

### CODE REVIEW

**Comment by sachaos** [comment]
Thank you!
I’ll review it.

File: lib/interface.go [hunk 1]
```diff
@@ -37,6 +39,22 @@ func SearchByID(repo Repository, id int) (data IDCarrier, err error) {
 	}
 }
 
+func SearchByIDPrefix(repo Repository, prefix string) (id int, err error) {
```
Comment (sachaos) [inline lib/interface.go]: I think that `CompleteItemIDByIDPrefix` is more appropriate for that function name.

File: lib/project.go [hunk 2]
```diff
@@ -24,3 +24,12 @@ func (a Projects) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
 func (a Projects) Less(i, j int) bool { return a[i].ID < a[j].ID }
 
 func (a Projects) At(i int) IDCarrier { return a[i] }
+
+func (a Projects) GetIDByName(name string) int {
+	for i := 0; i < a.Len(); i++ {
```
Comment (sachaos) [inline lib/project.go]: It is better to use `for range`.

```go
for _, pjt := range a {
    if pjt.Name == name {
        return pjt.GetID()
    }
    return 0
}
```

File: modify.go [hunk 3]
```diff
@@ -18,7 +18,7 @@ func Modify(c *cli.Context) error {
 	}
 
 	var err error
-	item_id, err := strconv.Atoi(c.Args().First())
+	item_id, err := todoist.SearchByIDPrefix(client.Store.Items, c.Args().First())
```
Comment (sachaos) [inline modify.go]: It seems to prefer to operate data in `client` through `client` .

Comment (mthowe) [reply modify.go]: Thanks for your feedback! I will address the issues at the nearest opportunity. However, I'm not entirely sure what you mean with this particular comment. Could you elaborate a bit or provide an example?

Comment (sachaos) [reply modify.go]: Implement `SearchByIDPrefix` method on `client`, then change the name. (https://github.com/sachaos/todoist/pull/35#discussion_r159706248)

e.g.

```go
item_id_prefix := c.Args().First()
item_id, err := client.CompleteItemIDByPrefix(item_id_prefix)
```

Comment (mthowe) [reply modify.go]: Ok, that's what I was thinking you meant but wanted to make sure. Thanks!

**Comment by mthowe** [comment]
The previously requested changes have been made and added to the PR.

**Comment by sachaos** [comment]
@mthowe Thank you so much!

**Pull request merged**